jdaviz/core/template_mixin.py

```diff
@@ -122,16 +131,125 @@ def app(self):
     def hub(self):
         return self._plugin.hub
 
+    @property
+    def viewer_dicts(self):
+        def _dict_from_viewer(viewer, viewer_item):
+            d = {'viewer': viewer, 'id': viewer_item['id']}
+            if viewer_item.get('reference') is not None:
+                d['reference'] = viewer_item['reference']
+                d['label'] = viewer_item['reference']
+            else:
+                d['reference'] = None
+                d['label'] = viewer_item['id']
+            return d
+
+        return [_dict_from_viewer(viewer, self.app._viewer_item_by_id(vid))
+                for vid, viewer in self.app._viewer_store.items()
+                if viewer.__class__.__name__ != 'MosvizTableViewer']
```

Comment:
Would we ever want to subclass `MosvizTableViewer`? If so, `isinstance` is safer maybe?

```suggestion
                if not isinstance(viewer, MosvizTableViewer)
```

Reply:
This would cause a loop in the import statements (I thought I left a comment, but maybe that was in another case of the same thing)

Reply:
Oh, ok.